### ISSUE TITLE
property does not exists on type never when using "in" operator

### ISSUE THREAD

**Issue by Dimous** [Microsoft/TypeScript#20590]
**TypeScript Version:** starting from 2.7.0-dev.20171207

**Code**
I'm making a shim for HTMLVideoElement. If "srcObject" does not exist, it uses "src".

```ts
    public setVideoSource(html_video_element: HTMLVideoElement, media_stream: MediaStream): void {
        if ("srcObject" in html_video_element) html_video_element.srcObject = media_stream;
        else
            html_video_element.src = URL.createObjectURL(media_stream); // property "src" does not exist on type never 
    }
```
at the same time it's ok if:

```ts
    public setVideoSource(html_video_element: HTMLVideoElement, media_stream: MediaStream): void {
        if (html_video_element.srcObject) html_video_element.srcObject = media_stream;
        else
            html_video_element.src = URL.createObjectURL(media_stream); // no error 
    }
```

**Expected behavior:**
no error

**Actual behavior:**
error

**Comment by Dimous** [comment]
It's also ok if hasOwnProperty("srcObject")

**Comment by andy-ms** [comment]
Duplicate of #20583

**Comment by weswigham** [comment]
@andy-ms This is requesting the opposite of #20583. #20583 says that we should make this an error for truthiness guards (so there's an error in both cases here), while this issue lists the expected behavior as there being an error in neither case. We briefly talked about this at our last design meeting, but it might be worth revisiting.
